CONTRIBUTING.txt

```diff
@@ -0,0 +1,98 @@
+
+1. Goals
+
+    The FreeCAD Contribution Process is expressed here with the following specific goals in mind:
+
+    1. To provide transparency and fairness in the contribution process.
+    2. To allow useful contributions to be included as quickly as possible.
+    3. To preserve and improve the code quality while encouraging appropriate experimentation and risk-taking.
+    4. To minimize dependence on individual contributors by encouraging a large pool of active contributors.
+    5. To be inclusive of many viewpoints and to harness a diverse set of skills.
+    6. To provide an encouraging environment where contributors learn and improve their skills.
+    7. To protect the free and open nature of the FreeCAD project.
+
+2. Fundamentals
+
+    1. FreeCAD uses the git distributed revision control system.
+    2. Source code for the main application and related subprojects is hosted on github.com in the FreeCAD organization, hereafter referred to as 'the platform'.
+    3. Problems are discrete, well-defined limitations or bugs.
+    4. FreeCAD uses the platform's issue-tracking system to track problems.
+    5. Patches are sets of code changes that resolve a single problem.
+    6. FreeCAD uses the Pull Request workflow for evaluating and accepting patches.
+
+
+3. Roles
+    1. "User": An member of the wider FreeCAD community who uses the software.
+    2. "Contributor":  A person who submits a patch that resolves a previously identified problem. Contributors do not have commit access to the repository unless they are also maintainers. Everyone, without distinction or discrimination, SHALL have an equal right to become a Contributor.
+    3. "Maintainer": A person who merges patches. Maintainers may or may not be Contributors. Their role is to enforce the process. Maintainers have commit access to the repository.
+    4. "Administrator": Administrators have additional authority to maintain the list of designated maintainers.
+
+4. Licensing, Ownership, and Credit
+    1. FreeCAD uses the Lesser General Public License, version 2, or superior (LGPL2+) for all source code.  Additional details can be found in the LICENSE file.
+    2. All contributions to FreeCAD SHALL use the same license as the project.
+    3. All contributions are owned by their authors unless assigned to another.
+    4. FreeCAD does not have a mandatory copyright assignment policy.
+    5. A Contributor who wishes to be identified in the Credits section of the application "About" dialog is responsible for identifying themselves appropriately by notifying an administrator or maintainer.
+
+5. Contribution Requirements
+
+    1. Patches are submitted in the form of Pull Requests (PR).
+    2. Maintainers and Contributors MUST have a platform account and SHOULD use their real names or a well-known alias. If the platform alias differs from the account alias on the FreeCAD Forum, effort SHOULD be taken to avoid confusion.
```

Comment:
Formatting comments:
Be carefull with line endings. If a clause is in a list of "or' items or 'and' items the more correct line ending is ; and on the last but one is also the 'and' or 'or'. Periods isolate clauses.
Sub clause numbers should be of the form 1.2.3 or 1.a.ii or whatever. A clause cut or copied from the text otherwise loses its positional status.
Line lengths should conform to project default, currently 100 characters. Long lines are a pain.

If you like I can come back later in the process and do a reformat.

Reply:
> this goes against our actual practice and IMO against basic respect of privacy,

Why?  Aliases are still permitted.  We don't require anyone to disclose their true identity.  The policy doesn't even require them to use the same alias in both places.  It simply says they SHOULD take action to avoid confusion.

As far as the MUST part of the clause that requires them to have a platform account, I personally am adamant on that point.  It's impractical to have a process for code contributions that allows people to completely avoid the process and do something else.  If we start accepting .diff patches via email and forum posts, it's chaos.

If you feel otherwise, please submit a text change so that we can debate it fairly.

Reply:
At the very least the rule 48 is adressing two separate issues and shoud be split. If for no other reason than discussing them is clearer.

Reply:
I'll take another look at it and change it if I can make it clearer and not wordier.  I think it's basically just addressing the single issue of the author identification.  Is this a big deal or a minor concern for you?

Reply:
Just many years writing/adjudicating/debating rules alerting me that current wording is not ideal, and while not a huge deal, is so easy to fix it might as well be done.

Reply:
> Just many years writing/adjudicating/debating rules alerting me that current wording is not ideal, and while not a huge deal, is so easy to fix it might as well be done.

I'm confused.  Your previous comment indicated that you thought the two sentences in the clause were addressing two different issues and should be two separate clauses.  1) requirement to have a platform account  2) effort to avoid alias confusion.  

Now you're saying the wording isn't ideal.

In my mind, both statements are referring to the same big idea:  We want contributions identified with known people/accounts and not anonymous.

I'll split the clause into two and resolve the conversation.  If you want different wording, please propose text and reopen it.